google/cloud/storage/internal/async/object_descriptor_impl.h

```diff
@@ -73,35 +82,28 @@ class ObjectDescriptorImpl
   void MakeSubsequentStream() override;
 
  private:
+  using StreamManager = MultiStreamManager<ReadStream, ReadRange>;
+  using StreamIterator =
+      MultiStreamManager<ReadStream, ReadRange>::StreamIterator;
+
   std::weak_ptr<ObjectDescriptorImpl> WeakFromThis() {
     return shared_from_this();
   }
 
-  // This may seem expensive, but it is less bug-prone than iterating over
-  // the map with the lock held.
-  auto CopyActiveRanges(std::unique_lock<std::mutex> const&) const {
-    return streams_.back().active_ranges;
-  }
-
-  auto CopyActiveRanges() const {
-    return CopyActiveRanges(std::unique_lock<std::mutex>(mu_));
-  }
-
-  auto CurrentStream(std::unique_lock<std::mutex>) const {
-    return streams_.back().stream;
-  }
+  // Logic to ensure a background stream is always connecting.
+  void AssurePendingStreamQueued();
```

Comment:
`AssurePendingStreamQueued` is always called with a lock held. To aid future maintainers, consider changing its signature to:

```
void AssurePendingStreamQueued(std::unique_lock<std::mutex> const&);
```

This helps signal that calling this function should only be done with the mutex locked, even though the lock isn't referenced in the function implementation.